### PR TITLE
Issue 4975: Replace Loading Cache with Regular Cache

### LINKED ISSUE #4975: Dont use loading cache to load completable futures, instead load the values if futures are successful.
**Problem description**
We currently load completablefutures into the cache and if the future fails, we invalidate the cache entry as a callback. 
This is advantageous in that if multiple concurrent requests were made for same resource, a single request is made and its future is loaded in the cache which is returned to all callers. 
Though we only cache immutable records, this above approach still breaks the write after read consistency guarantee if the get call was made after create record but it got access to a future that was created before the create call. 

So we should cache the value after fetching it from the store. Even if it means multiple concurrent requests can result in multiple calls into the store until the value is found and cached, it will guarantee stronger consistency. 

**Problem location**
Cache.java

**Suggestions for an improvement**
Fix the issue

### PR DESCRIPTION
**Pull request by shiveshr**
Signed-off-by: Shivesh Ranjan <shivesh.ranjan@gmail.com>

**Change log description**  
Replace the cache implementation to not load the completablefuture into the cache as that can cause consistency issues.

**Purpose of the change**  
Fixes #4975 

**What the code does**  
Changes the cache to load the value. The store first retrieves the value for mutable records from the cache. and if not found, it calls the store and loads the value from the future. 
We dont try to optimize on multiple concurrent invocations to the loading function because that will result in the original problem of consistency check failure. 

**How to verify it**  
All existing tests should continue to pass.

### CODE REVIEW

**Comment by shiveshr** [comment]
@pbelgundi  System test has passed for this branch.

File: controller/src/main/java/io/pravega/controller/store/stream/Cache.java [hunk 1]
```diff
@@ -10,53 +10,42 @@
...
 
-import javax.annotation.ParametersAreNonnullByDefault;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.CompletionException;
 import java.util.concurrent.TimeUnit;
-import java.util.function.Function;
 
 /**
  * Cache for asynchronously retrieving and loading records from underlying store using the supplied loader.
  * This uses Guava's loading cache and takes a loader function for loading entries into the cache.
```
Comment (pbelgundi) [inline controller/src/main/java/io/pravega/controller/store/stream/Cache.java:19]: We don't use LoadingCache any more...

File: controller/src/main/java/io/pravega/controller/store/stream/Cache.java [hunk 2]
```diff
@@ -10,53 +10,42 @@
...
     
-    private final LoadingCache<CacheKey, CompletableFuture<VersionedMetadata<?>>> cache;
+    private final com.google.common.cache.Cache<CacheKey, VersionedMetadata<?>> cache;
 
-    public Cache(final Function<CacheKey, CompletableFuture<VersionedMetadata<?>>> loader) {
+    public Cache() {
         cache = CacheBuilder.newBuilder()
                             .maximumSize(MAX_CACHE_SIZE)
                             .expireAfterAccess(2, TimeUnit.MINUTES)
-                            .build(new CacheLoader<CacheKey, CompletableFuture<VersionedMetadata<?>>>() {
```
[the remark this answers is not in the document]
Comment (shiveshr) [reply controller/src/main/java/io/pravega/controller/store/stream/Cache.java:40]: in that case the cache would need to define the semantics of what we do when the cache load fails if the data is not present or there is a transient error.
this keeps the cache semantics cleaner and abstracted away from what the callers may load into it and how they deal with different errors.

[the remark this answers is not in the document]
Comment (pbelgundi) [reply controller/src/main/java/io/pravega/controller/store/stream/Cache.java:40]: Ok.

**Pull request merged**